src/sentry/models/relocation.py

```diff
@@ -0,0 +1,252 @@
+    # for all newly imported orgs, will receive emails with status updates as the relocation
+    # progresses, etc) over its lifetime and once it is completed.
+    #
+    # This is left NULL because we want to retain the ability to audit and rollback a `Relocation`
+    # even in the event of the owner's account being deleted, but that `NULL` should never be set at
+    # creation time.
+    owner = HybridCloudForeignKey("sentry.User", null=True, on_delete="SET_NULL")
+
+    # Unique ID for this import attempt. All assembled files in the remote filestore will be in a
+    # directory named after this UUID.
+    uuid = UUIDField(db_index=True, unique=True, default=default_guid)
+
+    # Possible values are in the the Stage enum.
+    step = models.SmallIntegerField(choices=Step.get_choices(), default=None)
+
+    # Possible values are in the the Status enum.
+    status = models.SmallIntegerField(
+        choices=Status.get_choices(), default=Status.IN_PROGRESS.value
+    )
+
+    # Organizations, identified by slug, which this relocation seeks to import, specified by the
+    # user as part of their relocation request. These slugs are relative to the import file, not
+    # their final value post-import, which may be changed to avoid collisions.
+    want_org_slugs = models.JSONField(default=list)
+
+    # Users, identified by username, which this relocation seeks to import - picked directly from
+    # the raw JSON file submitted by the user. These usernames are relative to the import file, not
+    # their final value post-import, which may be changed to avoid collisions.
+    want_usernames = models.JSONField(null=True)
+
+    # The last status for which we have notified the user. It is `None` by default, to indicate that
+    # we have not yet sent the user a "your relocation is in progress" email.
+    latest_notified = models.SmallIntegerField(choices=Step.get_choices(), null=True, default=None)
+
+    # The last task started by this relocation. Because tasks for a given relocation are always
+    # attempted sequentially, and never in concurrently (that is, there is always at most one task
+    # per relocation running at a given time), we can be sure that this is globally accurate.
+    latest_task = models.CharField(max_length=64, default="")
+
+    # Number of attempts by the latest task.
+    latest_task_attempts = models.SmallIntegerField(default=0)
+
+    # If the relocation failed, provide a user-legible reason why that happened.
+    failure_reason = models.CharField(max_length=256, null=True, default=None)
+
+    __repr__ = sane_repr("owner", "uuid")
+
+    class Meta:
+        app_label = "sentry"
+        db_table = "sentry_relocation"
+
+
+@region_silo_only_model
+class RelocationFile(DefaultFieldsModel):
+    """
+    A `RelocationFile` is an association between a `Relocation` and a `File`.
+
+    This model should be created in an atomic transaction with the `Relocation` and `File` it points
+    to.
```

Comment:
Ah, that makes sense. I can either group as suggested above, or put all relocations into `files`. Any reason to prefer one over the other?

Reply:
I think our two leading options are none of the relocation tables in `files`, or all of them.  Having none means you can do transactions on relocations models, but not the files. And all of the tables alongside files means you can do transactions on everything at once. 

We have a few other subsystems that operate on files outside of transactions so that path will work well. There are a few debug symbol related models located alongside files so that path will also work.

I think having the transactional boundaries across files + relocations makes it worthwhile co-locating everything in `files`.

Reply:
Sounds good, I'll cook up those PRs now and change this one back to using `relocation` instead of `relocation_id` on `RelocationFile`. One thing to note: the future `Relocation*ImportChunk` models discussed in the design doc, but not yet included here, will need to be co-located with whatever model they are importing, so we may need more of those. Ex something like `RelocationRegionalFileImportChunk`, which would then be on the `files` connection so that we can keep that transaction atomic, and so on. But I think its fine for those models to be across a db boundary from the `Relocation*` models in this PR, since they will not need to be modified atomically together.